drivers/ethernet/eth_stm32_hal.c

```diff
@@ -720,8 +720,11 @@ static void rx_thread(void *arg1, void *unused1, void *unused2)
 	__ASSERT_NO_MSG(dev_data != NULL);
 
 	while (1) {
-		res = k_sem_take(&dev_data->rx_int_sem,
-			K_MSEC(CONFIG_ETH_STM32_CARRIER_CHECK_RX_IDLE_TIMEOUT_MS));
+		res = k_sem_take(
+			&dev_data->rx_int_sem,
+			COND_CODE_1(IS_ENABLED(CONFIG_ETH_STM32_CARRIER_CHECK),
```

Comment:
```suggestion
			COND_CODE_1(CONFIG_ETH_STM32_CARRIER_CHECK,
```

no need for ``IS_ENABLED``